gatorgrade/output/output.py

```diff
@@ -0,0 +1,127 @@
+"""Run checks and display whether each has passed or failed."""
+
+import subprocess
+from pathlib import Path
+from typing import List, Union
+import gator
+import rich
+from gatorgrade.input.checks import ShellCheck, GatorGraderCheck
+from gatorgrade.output.check_result import CheckResult
+
+# Disable rich's default highlight to stop number coloring
+rich.reconfigure(highlight=False)
+
+
+def _run_shell_check(check: ShellCheck) -> CheckResult:
+    """Run a shell check.
+
+    Args:
+        check: The shell check to run.
+
+    Returns:
+        The result of running the shell check as a CheckResult.
+    """
+    result = subprocess.run(
+        check.command,
+        shell=True,
+        check=False,
+        timeout=300,
+        stdout=subprocess.DEVNULL,
+        stderr=subprocess.DEVNULL,
+    )
+    passed = result.returncode == 0
+    diagnostic = "" if passed else f'The command "{check.command}" failed'
+    return CheckResult(
+        passed=passed, description=check.description, diagnostic=diagnostic
+    )
+
+
+def _run_gg_check(check: GatorGraderCheck) -> CheckResult:
+    """Run a GatorGrader check.
+
+    Args:
+        check: The GatorGrader check to run.
+
+    Returns:
+        The result of running the GatorGrader check as a CheckResult.
+    """
+    try:
+        result = gator.grader(check.gg_args)
+        passed = result[1]
+        description = result[0]
+        diagnostic = result[2]
+    # If arguments are formatted incorrectly, catch the exception and
+    # return it as the diagnostic message
+    # Disable pylint to catch any type of exception thrown by GatorGrader
+    except Exception as command_exception:  # pylint: disable=W0703
+        passed = False
+        description = f'Invalid GatorGrader check: "{" ".join(check.gg_args)}"'
+        diagnostic = f'"{command_exception.__class__}" thrown by GatorGrader'
+    return CheckResult(passed=passed, description=description, diagnostic=diagnostic)
+
+
+def run_checks(checks: List[Union[ShellCheck, GatorGraderCheck]]) -> None:
+    """Run shell and GatorGrader checks and display whether each has passed or failed.
+
+        Also, print a list of all failed checks with their diagnostics and a summary message that
+        shows the overall fraction of passed checks.
+
+    Args:
+        checks: The list of shell and GatorGrader checks to run.
+    """
+    results = []
+    for check in checks:
+        result = None
+
+        if isinstance(check, ShellCheck):
+            result = _run_shell_check(check)
+        elif isinstance(check, GatorGraderCheck):
+            result = _run_gg_check(check)
+
+        if result is not None:
+            result.print()
+            results.append(result)
+
+    failed_results = list(filter(lambda result: not result.passed, results))
+    if (
+        len(failed_results) > 0
+    ):  # Only print failures list if there are failures to print
```

Comment:
The comment should go above the `if` statement so that `black` does not wrap the statement.